File: sdkwork-ai-sdk-typescript/src/api/batches.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { BatchCreateRequest, BatchListResponse, BatchResponse } from '../types';


export class BatchesApi2 {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Cancel batch */
  async cancelBatch(batch_id: string | number): Promise<BatchResponse> {
    return this.client.post<BatchResponse>(aiApiPath(`/v1/batches/${batch_id}/cancel`));
  }

/** List batches */
  async listBatches(params?: QueryParams): Promise<BatchListResponse> {
    return this.client.get<BatchListResponse>(aiApiPath(`/v1/batches`), params);
  }

/** Create batch */
  async createBatch(body: BatchCreateRequest): Promise<BatchResponse> {
    return this.client.post<BatchResponse>(aiApiPath(`/v1/batches`), body);
  }

/** Get batch */
  async getBatch(batch_id: string | number): Promise<BatchResponse> {
    return this.client.get<BatchResponse>(aiApiPath(`/v1/batches/${batch_id}`));
  }
}

export function createBatchesApi2(client: HttpClient): BatchesApi2 {
  return new BatchesApi2(client);
}

export { BatchesApi2 as BatchesApi, createBatchesApi2 as createBatchesApi };

File: sdkwork-ai-sdk-flutter/lib/src/api/batches.dart
import '../http/client.dart';
import '../models.dart';

class BatchesApi {
  final HttpClient _client;
  
  BatchesApi(this._client);

  /// Cancel batch
  Future<BatchResponse?> cancelBatch(String batch_id) async {
    final response = await _client.post(ApiPaths.aiPath('/batches/${batch_id}/cancel'));
    return response is BatchResponse ? response : null;
  }

  /// List batches
  Future<BatchListResponse?> listBatches(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/batches'), params: params);
    return response is BatchListResponse ? response : null;
  }

  /// Create batch
  Future<BatchResponse?> createBatch(BatchCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/batches'), body: body, contentType: 'application/json');
    return response is BatchResponse ? response : null;
  }

  /// Get batch
  Future<BatchResponse?> getBatch(String batch_id) async {
    final response = await _client.get(ApiPaths.aiPath('/batches/${batch_id}'));
    return response is BatchResponse ? response : null;
  }
}
